src/Instrumentation/Symfony/src/HttpClientInstrumentation.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Symfony;
+
+use OpenTelemetry\API\Common\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Common\Instrumentation\Globals;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\Propagation\ArrayAccessGetterSetter;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+final class HttpClientInstrumentation
+{
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation('io.opentelemetry.contrib.php.symfony_http');
+
+        hook(
+            HttpClientInterface::class,
+            'request',
+            pre: static function (
+                HttpClientInterface $client,
+                array $params,
+                string $class,
+                string $function,
+                ?string $filename,
+                ?int $lineno,
+            ) use ($instrumentation): array {
+                /** @psalm-suppress ArgumentTypeCoercion */
+                $builder = $instrumentation
+                    ->tracer()
+                    ->spanBuilder(\sprintf('HTTP %s', $params[0]))
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::HTTP_URL, (string) $params[1])
+                    ->setAttribute(TraceAttributes::HTTP_METHOD, $params[0])
+                    ->setAttribute(TraceAttributes::CODE_FUNCTION, $function)
+                    ->setAttribute(TraceAttributes::CODE_NAMESPACE, $class)
+                    ->setAttribute(TraceAttributes::CODE_FILEPATH, $filename)
+                    ->setAttribute(TraceAttributes::CODE_LINENO, $lineno);
+
+                $propagator = Globals::propagator();
+                $parent = Context::getCurrent();
+
+                $span = $builder
+                    ->setParent($parent)
+                    ->startSpan();
+
+                if (!isset($params[2]['headers'])) {
+                    $params[2]['headers'] = [];
+                }
+
+                $previousOnProgress = $params[2]['on_progress'] ?? null;
+
+                $params[2]['on_progress'] = static function (int $dlNow, int $dlSize, array $info) use (
```

Comment:
This is because of https://github.com/symfony/contracts/blob/main/HttpClient/HttpClientInterface.php#L75 ("Responses MUST be lazy")? It might need a short comment about what's going on here...

Reply:
exactly, I added one sentence about this implementation